src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -23,6 +23,7 @@ internal class SNIPacket : IDisposable, IEquatable<SNIPacket>
 
         private ArrayPool<byte>  _arrayPool = ArrayPool<byte>.Shared;
         private bool _isBufferFromArrayPool = false;
+        private ValueTask _asyncOperation;
```

Comment:
Why is this needed?  It appears to only ever be written to, not read.

Reply:
Was going to ask you about this 😄 

Is it necessary to hold a reference to prevent the Task potentially getting GC'd? Though I'd guess the Read/Write Op would be holding it?

Reply:
> Is it necessary to hold a reference to prevent the Task potentially getting GC'd? Though I'd guess the Read/Write Op would be holding it?

Not if the async operation is implemented correctly: otherwise, how could it ever complete the task if it didn't keep it alive?

Reply:
Sometimes I over think things 😉

Reply:
Done